plugins/inputs/postgresql/postgresql.go

```diff
@@ -139,15 +144,13 @@ func (p *Postgresql) accRow(row scanner, acc telegraf.Accumulator, columns []str
 			}
 		}
 	} else {
-		if _, err := dbname.WriteString("postgres"); err != nil {
+		database, err := p.GetConnectDatabase(tagAddress)
+		if err != nil {
+			return err
+		}
```

Comment:
Instead of doing the connection string parsing yourself I suggest to use the [`ParseConfig` function](https://pkg.go.dev/github.com/jackc/pgx/v5#ParseConfig) of the `pgx` library.